.travis.yml

```diff
@@ -116,6 +116,10 @@ install:
 after_success:
   - ci/upload-codecov.sh
   - ci/upload-docs.sh
+  - ci/dump-logs.sh
```

Comment:
Why do we also want to dump the logs on success?

Reply:
I added this to check that the command works, planning to revert it before submitting. The problem is, it's not working even so --
```
cat: bigtable/tests/emulator.log: No such file or directory
```
https://travis-ci.org/GoogleCloudPlatform/google-cloud-cpp/jobs/374609663

and the only way I know how to iterate on this is to commit something and wait for the build results from Travis online. There are some instructions on how to run a build locally, but travis-debug comes as a docker image and docker-in-a-docker is tedious.

Reply:
Running in a Docker image as described in [CONTRIBUTING.md](https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/master/CONTRIBUTING.md#using-docker-to-compile-and-test) gets your 99% of the environment reproduced.

The problem is that you are missing the `build-output/cached-${DISTRO}-${DISTRO_VERSION}` subdirectory.  The docker based builds compile in a subdirectory and run there:

https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/master/ci/build-docker.sh#L70

It might be best to print the log from inside that script, if we ever change the location of the build subdirectory you would not want to change the script too.

Reply:
Thanks, indeed `find` points to the logs on my local setup too, I just didn't think to search.

From what I can see in the frozen build, the script is interrupted when it times out. I could experiment with adding more signals to the trap call in https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/20f0084464e14abf05ce7b08cd9e4a947c413b18/bigtable/tools/run_emulator_utils.sh#L30
but I didn't find out yet what signal (if any) is being sent on timeouts

Reply:
If I was implementing Travis I would kill a process that went over its time allotment with `SIGKILL`, otherwise their users could trap the signal and continue to run for a while, something that abusers (think crypto miners) would love to do.  Or I would just shutdown the VM, in which case the kernel is not there to deliver any signals.  In other words, I would not try to catch those, probably futile.